src/braket/experimental/autoqasm/instructions/instructions.py

```diff
@@ -14,15 +14,26 @@
 
 """Non-unitary instructions that apply to qubits."""
 
+from __future__ import annotations
+
+from collections.abc import Iterable
 from typing import Any
 
+import oqpy
+
 from braket.experimental.autoqasm import program as aq_program
 from braket.experimental.autoqasm.instructions.qubits import _qubit
 from braket.experimental.autoqasm.types import QubitIdentifierType
 
 
 def _qubit_instruction(
-    name: str, qubits: list[QubitIdentifierType], *args: Any, is_unitary: bool = True
+    name: str,
+    qubits: Iterable[QubitIdentifierType],
+    *args: Any,
+    is_unitary: bool = True,
+    control: QubitIdentifierType | Iterable[QubitIdentifierType] | None = None,
+    control_state: str | None = None,
```

Comment:
BDK also supports lists and ints for control state (ie. `"100"` == `[1, 0, 0]` == `4`) any plans to do the same here?

Reply:
Good point. This should probably accept `BasisStateInput` and share the BDK logic as much as possible. I'll update this.